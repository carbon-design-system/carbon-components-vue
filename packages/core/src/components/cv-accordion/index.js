import CvAccordion from './cv-accordion';
import CvAccordionItem from './cv-accordion-item';
import CvAccordionSkeleton from './cv-accordion-skeleton';

export { CvAccordion, CvAccordionItem, CvAccordionSkeleton };
