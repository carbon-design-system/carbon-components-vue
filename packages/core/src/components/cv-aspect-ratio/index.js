import CvAspectRatio from './cv-aspect-ratio';
export { CvAspectRatio };
