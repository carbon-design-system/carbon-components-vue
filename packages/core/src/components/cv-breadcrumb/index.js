import CvBreadcrumb from './cv-breadcrumb';
import CvBreadcrumbItem from './cv-breadcrumb-item';
import CvBreadcrumbSkeleton from './cv-breadcrumb-skeleton';

export { CvBreadcrumb, CvBreadcrumbItem, CvBreadcrumbSkeleton };
