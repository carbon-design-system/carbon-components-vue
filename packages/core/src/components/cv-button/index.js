import CvButton from './cv-button';
import CvIconButton from './cv-icon-button';
import CvButtonSkeleton from './cv-button-skeleton';
import CvButtonSet from './cv-button-set';

export { CvButton, CvIconButton, CvButtonSkeleton, CvButtonSet };
