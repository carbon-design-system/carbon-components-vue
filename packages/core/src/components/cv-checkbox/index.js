import CvCheckbox from './cv-checkbox';
import CvCheckboxSkeleton from './cv-checkbox-skeleton';
export { CvCheckbox, CvCheckboxSkeleton };
