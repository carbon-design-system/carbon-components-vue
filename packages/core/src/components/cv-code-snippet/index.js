import CvCodeSnippet from './cv-code-snippet';
import CvCodeSnippetSkeleton from './cv-code-snippet-skeleton';
export { CvCodeSnippet, CvCodeSnippetSkeleton };
