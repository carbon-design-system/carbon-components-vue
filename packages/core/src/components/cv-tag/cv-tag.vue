<template>
  <span
    :class="[
      `cv-tag ${carbonPrefix}--tag`,
      `${carbonPrefix}--tag--${tagKind}`,
      {
        [`${carbonPrefix}--tag--filter`]: isFilter,
        [`${carbonPrefix}--tag--disabled`]: disabled,
      },
    ]"
    role="listitem"
    :title="title"
    @keydown.enter.stop.prevent="$emit('remove')"
    @keydown.space.prevent
    @keyup.space.prevent="$emit('remove')"
  >
    <span :class="`${carbonPrefix}--tag__label`">{{ label }}</span>
    <button
      v-if="isFilter"
      :class="`${carbonPrefix}--tag__close-icon`"
      :aria-label="clearAriaLabel"
      @click.stop.prevent="onRemove"
      :disabled="disabled"
    >
      <Close16 />
    </button>
  </span>
</template>

<script>
import Close16 from '@carbon/icons-vue/es/close/16';
import { carbonPrefixMixin } from '../../mixins';

const tagKinds = [
  'red',
  'magenta',
  'purple',
  'blue',
  'cyan',
  'teal',
  'green',
  'gray',
  'cool-gray',
  'warm-gray',
  'high-contrast',
];

export default {
  name: 'CvTag',
  mixins: [carbonPrefixMixin],
  components: { Close16 },
  props: {
    clearAriaLabel: { type: String, default: 'Clear filter' },
    disabled: Boolean,
    label: { type: String, required: true },
    kind: {
      type: String,
      default: tagKinds[0],
      validator(val) {
        if (val === 'filter' && process.env.NODE_ENV === 'development') {
          console.warn('DEPRECARTED: Prefer props.filter (bool)');
          return true;
        }
        return tagKinds.includes(val);
      },
    },
    filter: {
      type: Boolean,
      default: false,
    },
  },
  computed: {
    isFilter() {
      return this.filter || this.kind === 'filter';
    },
    tagKind() {
      return this.kind === 'filter' ? 'high-contrast' : this.kind;
    },
    title() {
      return this.isFilter ? this.clearAriaLabel : null;
    },
  },
  methods: {
    onRemove() {
      if (!this.disabled) {
        this.$emit('remove');
      }
    },
  },
};
</script>
