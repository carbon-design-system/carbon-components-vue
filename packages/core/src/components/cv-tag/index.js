import CvTag from './cv-tag';
import CvTagSkeleton from './cv-tag-skeleton';
export { CvTag, CvTagSkeleton };
