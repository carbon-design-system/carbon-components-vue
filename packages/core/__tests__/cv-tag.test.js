// import { shallowMount as shallow, mount } from '@vue/test-utils';
import { testComponent, testInstance, awaitNextTick } from './_helpers';
const { shallowMount: shallow, mount, trigger } = awaitNextTick;
import { CvTag, CvTagSkeleton } from '@/components/cv-tag';
// import { settings as carbonSettings } from 'carbon-components';

// const carbonPrefix = carbonSettings.prefix;

describe('CvTag', () => {
  // ***************
  // PROP CHECKS
  // ***************
  testComponent.propsAreRequired(CvTag, ['label']);
  testComponent.propsHaveDefault(CvTag, ['kind', 'clearAriaLabel']);

  testComponent.propsAreType(CvTag, ['label', 'kind', 'clearAriaLabel'], String);
  testComponent.propsAreType(CvTag, ['disabled'], Boolean);

  // ***************
  // SNAPSHOT TESTS
  // ***************
  it('should render', async () => {
    const propsData = { kind: '', label: 'test' };
    const kinds = [
      'red',
      'magenta',
      'purple',
      'blue',
      'cyan',
      'teal',
      'green',
      'gray',
      'cool-gray',
      'warm-gray',
      'high-contrast',
    ];

    for (const kind of kinds) {
      propsData.kind = kind;
      const wrapper = await shallow(CvTag, { propsData });

      expect(wrapper.html()).toMatchSnapshot();
    }
  });

  // ***************
  // FUNCTIONAL TESTS
  // ***************
  testInstance.propStringIsRendered(CvTag, 'label', 'span');

  it('click on close icon emits remove', async () => {
    const propsData = { filter: true, label: 'test' };
    const wrapper = await mount(CvTag, { propsData });

    await trigger(wrapper.find('svg'), 'click');
    expect(wrapper.emitted().remove).toBeTruthy();
  });
});

describe('CvTagSkeleton', () => {
  // ***************
  // PROP CHECKS
  // ***************

  // ***************
  // SNAPSHOT TESTS
  // ***************
  it('Renders as expected', async () => {
    const wrapper = await shallow(CvTagSkeleton);

    expect(wrapper.html()).toMatchSnapshot();
  });
  // ***************
  // FUNCTIONAL TESTS
  // ***************
});
