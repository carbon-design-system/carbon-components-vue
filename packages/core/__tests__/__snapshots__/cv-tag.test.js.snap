// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CvTag should render 1`] = `
<span tabindex="0" role="listitem" class="cv-tag bx--tag bx--tag--red"><span class="bx--tag__label">test</span>
<!----></span>
`;

exports[`CvTag should render 2`] = `
<span tabindex="0" role="listitem" class="cv-tag bx--tag bx--tag--magenta"><span class="bx--tag__label">test</span>
<!----></span>
`;

exports[`CvTag should render 3`] = `
<span tabindex="0" role="listitem" class="cv-tag bx--tag bx--tag--purple"><span class="bx--tag__label">test</span>
<!----></span>
`;

exports[`CvTag should render 4`] = `
<span tabindex="0" role="listitem" class="cv-tag bx--tag bx--tag--blue"><span class="bx--tag__label">test</span>
<!----></span>
`;

exports[`CvTag should render 5`] = `
<span tabindex="0" role="listitem" class="cv-tag bx--tag bx--tag--cyan"><span class="bx--tag__label">test</span>
<!----></span>
`;

exports[`CvTag should render 6`] = `
<span tabindex="0" role="listitem" class="cv-tag bx--tag bx--tag--teal"><span class="bx--tag__label">test</span>
<!----></span>
`;

exports[`CvTag should render 7`] = `
<span tabindex="0" role="listitem" class="cv-tag bx--tag bx--tag--green"><span class="bx--tag__label">test</span>
<!----></span>
`;

exports[`CvTag should render 8`] = `
<span tabindex="0" role="listitem" class="cv-tag bx--tag bx--tag--gray"><span class="bx--tag__label">test</span>
<!----></span>
`;

exports[`CvTag should render 9`] = `
<span tabindex="0" role="listitem" class="cv-tag bx--tag bx--tag--cool-gray"><span class="bx--tag__label">test</span>
<!----></span>
`;

exports[`CvTag should render 10`] = `
<span tabindex="0" role="listitem" class="cv-tag bx--tag bx--tag--warm-gray"><span class="bx--tag__label">test</span>
<!----></span>
`;

exports[`CvTag should render 11`] = `
<span tabindex="0" role="listitem" class="cv-tag bx--tag bx--tag--high-contrast"><span class="bx--tag__label">test</span>
<!----></span>
`;

exports[`CvTagSkeleton Renders as expected 1`] = `<div class="cv-tag bx--tag bx--skeleton"></div>`;
